src/lib.rs

```diff
@@ -800,24 +799,23 @@ pub mod tests {
             }
             println!("Generated query set");
 
-            let opening_challenge = F::rand(rng);
             let proof = PC::batch_open(
                 &ck,
                 &polynomials,
                 &comms,
                 &query_set,
-                opening_challenge,
                 &rands,
                 Some(rng),
+                &mut PC::setup_oracle(),
```

Comment:
Here you can start the oracle directly in the test code rather than calling the setup_oracle() function

Reply:
Made as above